AirCasting/Notes/AddNote/AddNoteViewModel.swift

```diff
@@ -4,18 +4,26 @@ import Foundation
 import Resolver
 
 class AddNoteViewModel: ObservableObject {
+    @Injected private var locationTracker: LocationTracker
     @Published var noteText = Strings.Commons.note
     
     private let notesHandler: NotesHandler
     private let exitRoute: () -> Void
+    private let trackLocation: Bool
     
-    init(sessionUUID: SessionUUID, exitRoute: @escaping () -> Void) {
+    init(sessionUUID: SessionUUID, withLocation: Bool, exitRoute: @escaping () -> Void) {
         self.exitRoute = exitRoute
         self.notesHandler = Resolver.resolve(NotesHandler.self, args: sessionUUID)
+        trackLocation = withLocation
     }
     
     func continueTapped() {
-        notesHandler.addNote(noteText: noteText)
+        if let latitude = locationTracker.locationManager.location?.coordinate.latitude,
+           let longitude = locationTracker.locationManager.location?.coordinate.longitude,
+           trackLocation {
+            notesHandler.addNote(noteText: noteText, latitude: latitude, longitude: longitude)
+        }
+        notesHandler.addNote(noteText: noteText, latitude: 20.0, longitude: 20.0)
```

Comment:
Here when you add one note you will continue to next step, so whenever the user add a new note - he wil add a proper one as well as the fake one. I think we should avoid that, so maybe we could use a guard here? If you prefer `if let` then please combine it with `return` and use `exitRoute()` in both cases. Still I would like to say that seems to be working really nice 😊 

Example:

`guard let latitude = locationTracker.locationManager.location?.coordinate.latitude,
              let longitude = locationTracker.locationManager.location?.coordinate.longitude,
              trackLocation else {
            notesHandler.addNote(noteText: noteText, latitude: 20.0, longitude: 20.0)
            exitRoute()
            return
        }
        notesHandler.addNote(noteText: noteText, latitude: latitude, longitude: longitude)
        exitRoute() `